File: web/app/features/site-path-rule/types/site-path-check-history.dto.ts
/// 子路由检测历史 DTO（前后端字段名约定）。
///
/// 镜像 SiteCheckHistoryDto，增加 ruleId / path / text_matched 以支撑路径维度展示。
///
/// 字段语义：
/// - ruleId：关联的路径规则 id
/// - path：探测路径（冗余自规则，规则删除后历史仍可读）
/// - checkedAt：探测发生的毫秒时间戳
/// - status：探测动作结果；UP=拿到响应，ERROR=探测本身失败（超时/连接失败等）
/// - httpStatus：HTTP 响应码；探测失败时为 null
/// - textMatched：KEYWORD 模式下是否命中关键字；其余模式为 null
/// - errorMessage：探测失败原因；成功时为 null；数据库字段限长 512
import type { CheckStatus } from '~/features/site/types/check-status'

export interface SitePathCheckHistoryDto {
  id: number
  siteId: number
  ruleId: number
  path: string
  checkedAt: number
  status: CheckStatus
  httpStatus: number | null
  textMatched: boolean | null
  errorMessage: string | null
}

File: server/src/main/java/com/siteguard/monitor/dto/SitePathCheckHistoryDTO.java
package com.siteguard.monitor.dto;

import com.siteguard.monitor.entity.CheckStatus;

/// 单条子路由检测历史输出 DTO。
/// status 表示请求是否完成；业务判定分别由 httpStatus/textMatched/jsonMatched 展示。
public record SitePathCheckHistoryDTO(
        Long id,
        Long siteId,
        Long ruleId,
        String path,
        Long checkedAt,
        CheckStatus status,
        Integer httpStatus,
        Boolean textMatched,
        Boolean jsonMatched,
        String jsonDetail,
        String errorMessage
) {
    /// 兼容旧的 HTTP_STATUS/KEYWORD 测试与调用点。
    public SitePathCheckHistoryDTO(Long id, Long siteId, Long ruleId, String path,
                                   Long checkedAt, CheckStatus status, Integer httpStatus,
                                   Boolean textMatched, String errorMessage) {
        this(id, siteId, ruleId, path, checkedAt, status, httpStatus, textMatched,
                null, null, errorMessage);
    }
}
